clappr/clappr-core
# Core - Make reset CSS configurable

## Summary

This PR creates the possibility to don't use `nested-reset` CSS mixin when the new option `resetStyle` is configured with the `false` value. One default value for `resetStyle` option is created to maintain the current behavior.

Also, the style injection is changed to allow the new feature.

The path pattern to access modules is updated to use the `src` alias on all files where is necessary to update the style injection.

## Changes

- `rollup.config.js`:
  - Config the `rollup-plugin-postcss` to enable inject style on the element itself instad the `head` page
- `player.js`:
  -  Create default value (`true`) for `resetStyle` option
- `core/public/style.scss`:
  -  Remove `nested-reset` CSS mixin
- `core/public/optional.scss`:
  - Add `nested-reset` CSS mixin
- `core.js`:
  -  Add `optional.scss` accordingly the `resetStyle` value
- `core.js`/`container.js`/`html5_video.js`/`html_img.js`/`no_op.js`:
  - Use src path alias to import modules;
  - Inject style on the element itself instead the `head` page;

## How to test

- Run public page;
- Check the `head` tag on the page;
  - Should not have any anonymous style of any player component;
- Check any player element;
  - Should have the style related on the same level;
- Check `core` element on the page;
  - Should have two style elements (`CoreStyle` and `OptionalStyle`);
- Set `resetStyle` option with the `false` value;
  - Should have only one style element (`CoreStyle`);

## A picture tells a thousand words

## Before this PR

![Screen Shot 2020-11-10 at 16 03 08](https://user-images.githubusercontent.com/5631063/98723399-11181680-2371-11eb-928b-5b802dce47d0.png)

![Screen Shot 2020-11-10 at 16 03 35](https://user-images.githubusercontent.com/5631063/98723428-1d03d880-2371-11eb-8474-e282d77b89ae.png)


## After this PR

![Screen Shot 2020-11-10 at 16 00 22](https://user-images.githubusercontent.com/5631063/98723347-03629100-2371-11eb-9c91-1cdb195c6db4.png)

#### With `resetStyle = true` 

![Screen Shot 2020-11-10 at 16 00 46](https://user-images.githubusercontent.com/5631063/98723361-09f10880-2371-11eb-8022-0df0357bf767.png)

#### With `resetStyle = false`

![Screen Shot 2020-11-10 at 16 01 24](https://user-images.githubusercontent.com/5631063/98723603-5fc5b080-2371-11eb-9ff2-27ce047e49be.png)
